cmd/lambda/leases/main.go

```diff
@@ -73,6 +76,23 @@ type messageBody struct {
 	Body    string `json:"Body"`
 }
 
+func userDetailsMiddleware(next http.Handler) http.Handler {
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		reqCtx, err := muxLambda.GetAPIGatewayContext(r)
+		if err != nil {
+			log.Printf("Failed to parse context object from request: %s", err)
+			api.WriteAPIErrorResponse(w,
+				errors.NewInternalServer("Internal server error", err),
+			)
+			return
+		}
+		user = Services.UserDetailer().GetUser(&reqCtx)
```

Comment:
This is the best way to do this?  Seems weird to store something outside of the middleware?  Is this function that should be in the API?

Reply:
Yeah, it's kind of weird. I was trying to avoid using a context based on some discussions we'd had. But I think this is a good use case for it.